src/foundry/collections/compendiumCollection.d.ts

```diff
@@ -0,0 +1,179 @@
+ * Each Compendium pack has its own associated instance of the CompendiumCollection class which contains its contents.
+ *
+ * @param metadata -  The compendium metadata, an object provided by game.data
+ */
+declare class CompendiumCollection<T extends Entity = Entity> extends DocumentCollection<T> {
+  /**
+   * The amount of time that Document instances within this CompendiumCollection are held in memory.
+   * Accessing the contents of the Compendium pack extends the duration of this lifetime.
+   */
+  static CACHE_LIFETIME_SECONDS: number;
+
+  /**
+   * The named game setting which contains Compendium configurations.
+   */
+  static CONFIG_SETTING: string;
+
+  /**
+   * Create a new Compendium Collection using provided metadata.
+   * @param metadata -  The compendium metadata used to create the new pack
+   * @param options -  Additional options which modify the Compendium creation request
+   */
+  static createCompendium(metadata: Compendium.Metadata, options?: Compendium.Settings): Promise<CompendiumCollection>;
+
+  constructor(metadata: Compendium.Metadata);
+
+  /**
+   * The compendium metadata which defines the compendium content and location
+   */
+  metadata: Compendium.Metadata;
+
+  /**
+   * A subsidiary collection which contains the more minimal index of the pack
+   */
+  index: foundry.utils.Collection<Compendium.IndexEntry>;
+
+  /**
+   * A debounced function which will clear the contents of the Compendium pack if it is not accessed frequently.
+   */
+  private _flush;
+
+  /**
+   * The canonical Compendium name - comprised of the originating package and the pack name
+   */
+  get collection(): string;
+
+  /**
+   * Access the compendium configuration data for this pack
+   */
+  get config(): Compendium.Settings;
+
+  /**
+   * Track whether the Compendium Collection is locked for editing
+   */
+  get locked(): boolean;
+
+  /**
+   * Track whether the Compendium Collection is private
+   */
+  get private(): boolean;
+
+  /**
+   * A convenience reference to the label which should be used as the title for the Compendium pack.
+   */
+  get title(): string;
+
+  /**
+   * Load the Compendium index and cache it as the keys and values of the Collection.
+   */
+  getIndex(): Promise<CompendiumCollection<T>['index']>;
+
+  /**
+   * Get a single Document from this Compendium by ID.
+   * The document may already be locally cached, otherwise it is retrieved from the server.
+   * @param id -              The requested Document id
+   * @returns     The retrieved Document instance
+   */
+  getDocument(id: string): Promise<T>;
+
+  /**
+   * Load multiple documents from the Compendium pack using a provided query object.
+   * @param query -           A database query used to retrieve documents from the underlying database
+   * @returns   The retrieved Document instances
+   */
+  getDocuments(query?: object): Promise<T[]>;
```

Comment:
We should add a TODO here to remember to type the query later, if it can be typed.